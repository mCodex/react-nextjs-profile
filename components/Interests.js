import React, { PureComponent, Fragment } from 'react';
import { List, Grid } from 'semantic-ui-react';
import { translate } from 'react-i18next';

import withData from '../hocs/DataProvider';

class Interests extends PureComponent {
  render() {
    const { data, t } = this.props;
    return (
      <Fragment>
        <Grid stackable>
          <List>
            <List.Header as="h2">{t('Interests')}</List.Header>
            {data.interests.map(interest =>
              <List.Item>{interest}</List.Item>)
            }
          </List>
        </Grid>
      </Fragment>
    );
  }
}

export default withData(translate(['common'])(Interests));
