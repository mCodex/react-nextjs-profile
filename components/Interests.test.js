import React from 'react';
import renderer from 'react-test-renderer';

import Interests from './Interests';

describe('Interests component', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<Interests />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
