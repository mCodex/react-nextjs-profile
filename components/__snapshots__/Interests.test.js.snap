// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Interests component renders correctly 1`] = `
<div
  className="ui stackable grid"
>
  <div
    className="ui list"
    role="list"
  >
    <h2
      className="header"
    >
      <span
        className={undefined}
        style={undefined}
      >
        interests
      </span>
    </h2>
    <div
      className="item"
      data-value={undefined}
      onClick={[Function]}
      role="listitem"
    >
      Javascript
    </div>
    <div
      className="item"
      data-value={undefined}
      onClick={[Function]}
      role="listitem"
    >
      NodeJS
    </div>
    <div
      className="item"
      data-value={undefined}
      onClick={[Function]}
      role="listitem"
    >
      Blockchain
    </div>
  </div>
</div>
`;
